app/services/articles/creator.rb

```diff
@@ -12,8 +13,28 @@ def self.call(*args)
     def call
       raise if RateLimitChecker.new(user).limit_by_action("published_article_creation")
 
-      tags = article_params[:tags]
+      article = save_article.decorate
+
+      if article.persisted?
+        NotificationSubscription.create(user: user, notifiable_id: article.id, notifiable_type: "Article", config: "all_comments")
+        Notification.send_to_followers(article, "Published") if article.published
+
+        dispatch_event(article)
+      end
+      article
+    end
+
+    private
+
+    attr_reader :user, :article_params, :event_dispatcher
+
+    def dispatch_event(article)
+      event_dispatcher.call("article_created", article)
```

Comment:
I noticed that sometimes you pass the decorated article, other times like here you don't.

Can't the dispatcher itself just call ".decorate" internally? or even the serializer. Decoration seem more like a presentation thing (ie. JSON serialization or HTML rendering)

Reply:
For now, It's always decorated for the persisted article. First, I placed the logic in the dispatcher, but that's not right in my opinion. The dispatcher shouldn't be aware of if the record should be decorated or not.
However, I like the approach of decorating inside the serializer. I'll try it out tomorrow.

Reply:
Agreed that the dispatcher shouldn't know, but not even the caller should know, decorating is about the presentation layer, in theory, maybe by moving decoration inside the serializer you're set

Reply:
I haven't found a way to decorate inside a serializer, so I put the logic to the adapter for now.
It could be a separate object later (to prepare/decorate the object for serializing) if we need more complex logic.